src/ros_filter.cpp

```diff
@@ -1975,27 +1997,39 @@ namespace RobotLocalization
       for (size_t j = 0; j < dimension; j++)
       {
         covariance(i, j) = arr[dimension * i + j];
+      }
+    }
 
+    for (size_t i = 0; i < dimension; i++)
```

Comment:
The diff is a little crazy here. I basically added the covariance volume check and remove the previous check that was applied on single elements of the covariance matrix looking for large values (not deltas).